spec/security-warnings-spec.js

```diff
@@ -78,6 +78,22 @@ describe('security warnings', () => {
     w.loadURL(`http://127.0.0.1:8881/base-page-security.html`)
   })
 
+  it('should not warn about Node.js integration with remote content from localhost', (done) => {
+    w = new BrowserWindow({
+      show: false,
+      webPreferences: {
+        nodeIntegration: true
+      }
+    })
+    w.webContents.once('console-message', (e, level, message) => {
+      expect(message).to.not.include('Node.js Integration with Remote Content')
+      done()
+    })
+    w.webContents.once('did-finish-load', () => done())
```

Comment:
You can add a `console.log('loaded')` in `base-page-security.html` and then end the test when it is printed.

```js
on('console-message', () => {
  expect(message).to.not.include('Node.js Integration with Remote Content')
  if (message === 'done')
    done()
})
```

(other test loading `base-page-security.html` may be affected so you probably need to create a new file)

Reply:
Thanks for the suggestion, but I don't think that will work. `console.log('loaded')` will  be executed _before_ the nodeIntegration check has a chance to run, resulting in the test always passing, even if the warning is shown.

We'd need some way to detect when the nodeIntegration check is finished, so we can pass knowing that a message wasn't shown.

Reply:
Can you have a try of `window.onload = () => console.log('loaded')`?

Reply:
Nice, that worked.

I assume the event handler in [`security-warning.ts`](https://github.com/electron/electron/blob/master/lib/renderer/security-warnings.ts#L280) gets registered before the one I added, so when the `onload` event fires, the nodeIntegration check is guaranteed to execute first.

We could've actually put the event handler in `base-page-security.html` without breaking existing tests since they only listen for the first console message and "loaded" will come after. However, I think it'd be safer to not pollute the base page, so I added `base-page-security-loaded-message.html`.